src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -222,13 +223,34 @@ case object TChar extends Type {
     .map(s => s.substring(0, 1))
 }
 
-abstract class TNumeric extends Type
+object TNumeric {
+  def parseResultType(types: Set[TNumeric]): Type = {
+    if (types(TDouble) || (types(TFloat) && (types(TInt) || types(TLong))))
+      TDouble
+    else if (types(TInt) && types(TLong))
+      TLong
```

Comment:
this return type order is what I'm doing.  if there's a double, return a double.  If there's an int and a long, return a long.  If there's a float and anything else, return double.  Otherwise, there's just 1 type and return that.